input/list.go

```diff
@@ -215,6 +224,7 @@ func (s *List) processListFile(f io.ReadCloser) {
 
 	go func() {
 		scanner := bufio.NewScanner(f)
+		defer f.Close()
```

Comment:
I seem to have missed this the first time.

`processListFile` is used 4 times in the code base:
 - once in case `s3` and `f` is closed by the caller.
 - once in case `http` and `f` is closed by the caller.
 - once in case `-` (for stdin) `f` isn't closed by the caller (and it's ok).
 - once in case `file` and `f` isn't closed by the caller (which is a bug).

I admit the API is ambiguous: since `f` is a `ReadCloser`, one expects to have to close it, however as it's the case with stdin we don't always want to do it.

I think that the best way to solve this is to change the `processListFile` prototype to make it accept an `io.Reader`, not even giving the possibliity for us to close so it becomes obvious that it's the caller responsibility to do it if needed.